pkg/client/linstor.go

```diff
@@ -414,35 +414,36 @@ func (s *Linstor) Attach(ctx context.Context, vol *volume.Info, node string) err
 	if err != nil {
 		return err
 	}
+
+	rc.Resource.Props[linstor.PropertyCreatedFor] = linstor.CreatedForTemporaryDisklessAttach
 	return s.client.Resources.Create(ctx, rc)
 }
 
 // Detach removes a volume from the node.
 func (s *Linstor) Detach(ctx context.Context, vol *volume.Info, node string) error {
+	log := s.log.WithFields(logrus.Fields{
+		"volume":     fmt.Sprintf("%+v", vol),
+		"targetNode": node,
+	})
+
 	res, err := s.client.Resources.Get(ctx, vol.ID, node)
-	if err == lapi.NotFoundError {
-		s.log.WithFields(logrus.Fields{
-			"volume":     fmt.Sprintf("%+v", vol),
-			"targetNode": node,
-		}).Info("volume does not even exist")
-		return nil
-	} else if err != nil {
-		return err
+	if err != nil {
+		// if the resource could not be found there is nothing to detach
+		return nil404(err)
 	}
 
-	s.log.WithFields(logrus.Fields{
-		"resource":   fmt.Sprintf("%+v", res),
-		"targetNode": node,
-	}).Info("detaching volume")
+	createdFor, ok := res.Props[linstor.PropertyCreatedFor]
+	if !ok || createdFor != linstor.CreatedForTemporaryDisklessAttach {
```

Comment:
hm, what about legacy compat? wouldn't that keep the ones that truly have been created for attach only around forever? do users really create intentional/manual diskless assignments in a CSI context? if not, then we could delete resources that don't have the property *and* are diskeless *and* are not tie breakers. Does LINSTOR expose that information if it was a created as tie breaker? maybe we should request such a "tag" from the LINSTOR folks?

All that said, I'm fine if you merge it like this, keeping some unnecessary diskless resources should not hurt that much, probably not worth the effort doing some legacy compat here

Reply:
Yeah, this bit is a compromise. From my tests, LINSTOR removes the `TIE_BREAKER` flag from a resource once it is InUse. So after the volume is attached, we have no way to differentiate a diskless resource created as a tie breaker and a diskless resource created by CSI.

While we could argue that is a bug/missing feature in LINSTOR, I'd much rather wait for a better API for this use case in LINSTOR. Basically I just want to tell LINSTOR: I need this resource on that node, and later:  I don't need this anymore. LINSTOR can then figure out what to do based on the autoplace configuration.

Reply:
True, I'm always in shifting responsibility towards LINSTOR :). jokes aside, you are obviously right, that should be actually handled by LINSTOR in some improved way